File: java/121324_S5_4335.java
/**
 * https://www.acmicpc.net/problem/4335
 */
import java.io.*;
import java.util.*;

class Main {
	
	@FunctionalInterface interface Function { void run(int n); }
	
	static Map<String, Function> funcMap = new HashMap<>() {{
		put("too high", Main::tooHigh);
		put("too low", Main::tooLow);
		put("right on", Main::rightOn);
	}};
	
	static boolean[] t = new boolean[10];
	
	static {
		Arrays.fill(t, true);
	}
	
	static void tooHigh(int n) {
		for (int i = n; i < 10; i++) {
			t[i] = false;
		}
	}
	static void tooLow(int n) {
		for (int i = 0; i <= n; i++) {
			t[i] = false;
		}
	}
	static void rightOn(int n) {
		System.out.println(t[n] ? "Stan may be honest": "Stan is dishonest");
		Arrays.fill(t, true);
	}
	
	public static void main(String[] args) throws IOException {
		try(var br = new BufferedReader(new InputStreamReader(System.in))) {
			while(true) {
				int n = Integer.parseInt(br.readLine());
				if (n == 0) break;
				funcMap.get(br.readLine()).run(n-1);
			}
		}
	}
}

File: python/121324_S5_4335.py
# https://www.acmicpc.net/problem/4335

def too_high(n):
    for i in range(n, 10):
        t[i] = False

def too_low(n):
    for i in range(n+1):
        t[i] = False

def right_on(n):
    if t[n]:
        print("Stan may be honest")
    else:
        print("Stan is dishonest")

func_map = {
    "too high": too_high,
    "too low": too_low,
    "right on": right_on
}

t = [True] * 10
while True:
    n = int(input())
    if n == 0:
        break;
    inp = input()
    func_map[inp](n-1)
    if inp == "right on":
        t = [True] * 10

